src/nlp/arrow_dataset.py

```diff
@@ -737,6 +775,7 @@ def set_format(
                 )
             )
 
+        format_kwargs.update(format_kwargs.pop("format_kwargs", {}))  # allow to use self.set_format(self.format)
```

Comment:
I had to add this line to make `format_kwargs` work when doing `self.set_format(self.format)`